rust/src/encodings/plain.rs

```diff
@@ -129,13 +130,16 @@ impl<'a> PlainDecoder<'a> {
             DataType::Boolean => bit_util::ceil(self.length, 8),
             _ => get_primitive_byte_width(self.data_type)? * self.length,
         };
-        let range = Range {
-            start: self.position,
-            end: self.position + array_bytes,
-        };
+        let mut start = self.position;
+        let end = self.position + array_bytes;
+        let mut ranges: Vec<Range<usize>> = vec![];
+        while start < end {
+            ranges.push(start..min(start + 1 * 1024 * 1024, end));
+            start += 1 * 1024 * 1024;
+        }
 
-        let data = self.reader.get_range(range).await?;
-        let buf: Buffer = data.into();
+        let data = self.reader.get_ranges(&ranges).await?;
```

Comment:
does the inner object reader do parallel reads?

Reply:
object_store crates doing each range in a async func, so with enough threads in tokio, it is parallel reads.

`object_store` implementation for the local file system is not high perf tho. We might need to re-implement the local SSD access later, maybe or not under the same interface. For one, it opens and closes the file descriptor for EVERY READ. quite expensive overhead, and destroys page cache. 

It is cheaper / faster than read a large chunk from local SSD, so is cheaper than S3 (which is slower than SSD).

We might want to smart about the chunking tho. make sure it does not have too many small chunks. Maybe I should limit a upper bound of # of chunks, i.e., 32?